examples/generate/src/main.rs

```diff
@@ -1,12 +1,8 @@
-#[allow(warnings, unused)]
-mod contract {
-    include!(concat!(env!("OUT_DIR"), "/rust_coin.rs"));
-}
-
-use crate::contract::RustCoin;
 use web3::api::Web3;
 use web3::transports::Http;
 
+include!(concat!(env!("OUT_DIR"), "/rust_coin.rs"));
```

Comment:
Is this what we need to use instead of the contract macro now? I don't see where OUT_DIR is defined and am assuming `/rust_coin.rs` is generated? How does the generation happen?

Reply:
`OUT_DIR` is defined by Cargo, it is used as the build output directory for non-rust things. Specifically, if you were building and statically linking C modules into a rust crate, you would output them here so they get pulled in at link time. It is commonly used as a dumping ground for generated code by many `build.rs` code generation libraries (like `bindgen` for example - they are a good one to follow cause they are maintained by the Rust team now so it tends to be pretty "rusty").

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

> `OUT_DIR` - If the package has a build script, this is set to the folder where the build script should place its output. See below for more information.